src/include/uc_utils.hpp

```diff
@@ -23,6 +27,48 @@ struct UCType {
 	vector<UCType> children;
 };
 
+/**
+ * Manages AWS temporary credentials caching for Unity Catalog tables.
+ * Provides thread-safe credential caching with expiration checking.
+ */
+class UCTableCredentialManager {
+public:
+	/**
+	 * Get the singleton instance of the credential manager.
+	 */
+	static UCTableCredentialManager &GetInstance();
```

Comment:
A singleton is still just a global variable, but with extra steps ;)

Can we maybe store the UCTableCredentialManager in the catalog instead?